google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -144,8 +146,31 @@ public static CloudStorageFileSystem forBucket(String bucket, CloudStorageConfig
     checkArgument(
         !bucket.startsWith(URI_SCHEME + ":"), "Bucket name must not have schema: %s", bucket);
     checkNotNull(config);
-    return new CloudStorageFileSystem(
-        new CloudStorageFileSystemProvider(config.userProject()), bucket, config);
+    return new CloudStorageFileSystem(getCloudStorageFileSystemProvider(config), bucket, config);
+  }
+
+  /**
+   * Creates new file system provider instance for {@code CloudStorageFileSystemProvider}.
+   *
+   * @param config CloudStorageConfiguration to initialize {@code CloudStorageFileSystemProvider}
+   *     object
+   * @param storageOptions
+   * @return {@code CloudStorageFileSystemProvider} object
+   */
+  private static CloudStorageFileSystemProvider getCloudStorageFileSystemProvider(
+      CloudStorageConfiguration config, StorageOptions... storageOptions) {
+    CloudStorageFileSystemProvider cloudStorageFileSystemProvider =
+        (storageOptions.length >= 1)
+            ? new CloudStorageFileSystemProvider(config.userProject(), storageOptions[0])
+            : new CloudStorageFileSystemProvider(config.userProject());
+    if (!FILE_SYSTEM_CONFIG.isEmpty()
```

Comment:
it's not needed to check that FILE_SYSTEM_CONFIG is empty

Reply:
Done